core/src/loader.rs

```diff
@@ -122,6 +137,56 @@ pub fn load_associated_token_account(
     Ok(())
 }
 
+/// Loads the account as a token account, ensuring it is correctly linked to the specified mint and is owned by the expected token program.
+///
+/// This function performs the following checks:
+/// 1. Verifies that the `token_account` is associated with the expected SPL Token program.
+/// 2. Checks that the `token_account` is not empty and contains valid data.
+/// 3. Confirms that the `token_account` is linked to the specified `mint`, ensuring it is the correct token account for that mint.
+///
+/// # Arguments
+/// * `token_account` - The account to load the token account from
+/// * `mint` - The mint of the token account
+/// * `token_program_info` - The token program of the token account
+///
+/// # Returns
+/// * `Result<(), ProgramError>` - The result of the operation
+///
+/// # Errors
+/// This function will return an error in the following cases:
+/// * `ProgramError::InvalidAccountOwner` - If the `token_account` is not owned by the expected SPL Token program.
+/// * `ProgramError::InvalidAccountData` - If the `token_account` data is empty or if the mint associated with the `token_account` does not match the provided `mint`.
+pub fn load_token_account(
+    token_account: &AccountInfo,
+    mint: &Pubkey,
```

Comment:
should add owner in here too

Reply:
Pass the expect owner to this loader to check `token_account` owner is equal to expect owner.